rye/src/cli/init.rs

```diff
@@ -82,193 +85,54 @@ pub struct Args {
+
+    /// Generate an executable project.
+    #[arg(long)]
+    script: bool,
 }
 
-/// A Python module implemented in Rust.
-#[pymodule]
-fn _lowlevel(_py: Python, m: &PyModule) -> PyResult<()> {
-    m.add_function(wrap_pyfunction!(hello, m)?)?;
-    Ok(())
+enum TemplateChoice {
+    Lib,
+    Script,
 }
-"#;
 
-/// Template for the __init__.py
-const RUST_INIT_PY_TEMPLATE: &str = r#"from {{ name_safe }}._lowlevel import hello
-__all__ = ["hello"]
+/// The pyproject.toml template
+const TOML_TEMPLATE: &str = include_str!("../templates/pyproject.toml");
 
-"#;
+/// The template for the README.md.
+const README_TEMPLATE: &str = include_str!("../templates/README.md");
 
-/// Template for the Cargo.toml
-const CARGO_TOML_TEMPLATE: &str = r#"[package]
-name = {{ name }}
-version = "0.1.0"
-edition = "2021"
+/// The template for the LICENSE.txt.
+const LICENSE_TEMPLATE: &str = include_str!("../templates/LICENSE.txt");
 
-# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
-[lib]
-name = {{ name_safe }}
-crate-type = ["cdylib"]
+/// Template for the __init__.py when --lib is specified.
+const INIT_PY_LIB_TEMPLATE: &str = include_str!("../templates/lib/default/__init__.py");
 
-[dependencies]
-pyo3 = "0.19.0"
+/// Template for the __init__.py when --script is specified.
+const INIT_PY_BIN_TEMPLATE: &str = include_str!("../templates/script/default/__init__.py");
 
-"#;
+const INIT_PY_BIN_MAIN_TEMPLATE: &str = include_str!("../templates/script/default/__main__.py");
 
-/// Template for fresh gitignore files
-const GITIGNORE_TEMPLATE: &str = r#"# python generated files
-__pycache__/
-*.py[oc]
-build/
-dist/
-wheels/
-*.egg-info
+/// Template for the lib.rs when using the maturin build system.
+const LIB_RS_TEMPLATE: &str = include_str!("../templates/lib/maturin/lib.rs");
 
-{%- if is_rust %}
-# Rust
-target/
-{%- endif %}
+/// Template for the __init__.py when using the maturin build system.
+const RUST_INIT_PY_TEMPLATE: &str = include_str!("../templates/lib/maturin/__init__.py");
 
-# venv
-.venv
+/// Template for the Cargo.toml.
+const CARGO_TOML_TEMPLATE: &str = include_str!("../templates/lib/maturin/Cargo.toml");
 
-"#;
+/// Template for fresh gitignore files.
+const GITIGNORE_TEMPLATE: &str = include_str!("../templates/gitignore");
 
 /// Script used for setup.py setup proxy.
-const SETUP_PY_PROXY_SCRIPT: &str = r#"
-import json, sys
-from pathlib import Path
-from tempfile import TemporaryDirectory
-
-def setup(**kwargs) -> None:
-    print(json.dumps(kwargs), file=sys.stderr)
-
-if __name__ == "setuptools":
-    _setup_proxy_module = sys.modules.pop("setuptools")
-    _setup_proxy_cwd = sys.path.pop(0)
-    import setuptools as __setuptools
-    sys.path.insert(0, _setup_proxy_cwd)
-    sys.modules["setuptools"] = _setup_proxy_module
-    def __getattr__(name):
-        return getattr(__setuptools, name)
-"#;
+const SETUP_PY_PROXY_SCRIPT: &str = include_str!("../templates/setuptools.py");
```

Comment:
I don't think this should be in `templates` since it's not a template.

Reply:
Fair enough will do.